src/CarbonAware.WebApi/test/integrationTests/mocks/WattTimeDataSourceMocker.cs

```diff
@@ -0,0 +1,53 @@
+﻿using CarbonAware.DataSources.Configuration;
+using CarbonAware.Tools.WattTimeClient;
+using CarbonAware.Tools.WattTimeClient.Configuration;
+using CarbonAware.Tools.WattTimeClient.Model;
+using Microsoft.AspNetCore.Mvc.Testing;
+using Microsoft.Extensions.DependencyInjection;
+using WireMock.Server;
+
+namespace CarbonAware.WebApi.IntegrationTests;
+public class WattTimeDataSourceMocker : IDataSourceMocker
+{
+    protected WireMockServer _server;
+    private readonly object _dataSource = DataSourceType.WattTime;
+
+    internal WattTimeDataSourceMocker()
+    {
+        _server = WireMockServer.Start();
+        WattTimeServerMocks.WattTimeServerSetupMocks(_server);
+    }
+
+    public void SetupDataMock(DateTime start, DateTime end, string location)
```

Comment:
I think that either this method should own generating new datapoints based on a set of parameters (and not use default data at all) or let the test define its own data and just pass it along to the mock server (my personal preference).

This is currently used by the `best` endpoint test.  A good data set for that endpoint would include at least two points.  This method is not flexible enough to able to do that at present.

Reply:
As a bonus it would reveal what I learned this morning... that the `/best` endpoint in fact is broken 🙃

Reply:
We update the test to 'travel' the location through the integration-mock, and we opened a bug to track the `/best` issue, and are leaving writing an improved integration test for when it's fixed.